app/knight.py

```diff
@@ -0,0 +1,53 @@
+class Knight:
+    def __init__(self, config: dict) -> None:
+        self.name = config["name"]
+        self.power = self.apply_power(config)
+        self.protection = self.apply_armour(config)
+        self.hp = self.apply_hp(config)
+
+    @staticmethod
+    def apply_armour(config) -> int:
+        protection = 0
+
+        if config["potion"] is not None:
+            if "protection" in config["potion"]["effect"]:
+                protection += config["potion"]["effect"]["protection"]
```

Comment:
The code assumes that 'potion' and 'effect' keys exist and are structured correctly. Consider adding checks to ensure these keys exist to avoid potential KeyErrors.